mycroft/util/parse.py

```diff
@@ -164,6 +222,580 @@ def extractnumber_en(text):
+                                ):
+                                if (
+                                        wordNextNext == "afternoon" or
+                                        (len(words) > words.index(wordNextNext) + 1 and  # noqa
+                                        words[words.index(wordNextNext)+1] == "afternoon")  # noqa
+                                    ):
+                                    remainder = "pm"
+                                if (
+                                        wordNextNext == "evening" or
+                                        (len(words) > (words.index(wordNextNext) + 1) and  # noqa
+                                        words[words.index(wordNextNext)+1] == "evening")  # noqa
+                                    ):
+                                    remainder = "pm"
+                                if (
+                                        wordNextNext == "morning" or
+                                        (len(words) > words.index(wordNextNext) + 1 and  # noqa
+                                        words[words.index(wordNextNext)+1] == "morning")  # noqa
+                                    ):
+                                    remainder = "am"
+                        if timeQualifier != "":
+                            military = True
+                    else:
+                        isTime = False
+
+            strHH = int(strHH) if strHH else 0
+            strMM = int(strMM) if strMM else 0
+            strHH = strHH+12 if remainder == "pm" and strHH < 12 else strHH
+            strHH = strHH-12 if remainder == "am" and strHH >= 12 else strHH
+            if strHH > 24 or strMM > 59:
+                isTime = False
+                used = 0
+            if isTime:
+                hrAbs = strHH * 1
+                minAbs = strMM * 1
+                used += 1
+        if used > 0:
+            # removed parsed words from the sentence
+            for i in range(used):
+                words[idx + i] = ""
+
+            if wordPrev == "o" or wordPrev == "oh":
+                words[words.index(wordPrev)] = ""
+
+            if wordPrev == "early":
+                hrOffset = -1
+                words[idx-1] = ""
+                idx -= 1
+            elif wordPrev == "late":
+                hrOffset = 1
+                words[idx-1] = ""
+                idx -= 1
+            if idx > 0 and wordPrev in markers:
+                words[idx-1] = ""
+            if idx > 1 and wordPrevPrev in markers:
+                words[idx-2] = ""
+
+            idx += used-1
+            found = True
+
+    # check that we found a date
+    if (
+            not found and
+            (
+                datestr == "" and timeStr == "" and
+                yearOffset == 0 and monthOffset == 0 and
+                dayOffset is False and hrOffset == 0 and
+                hrAbs == 0 and minOffset == 0 and
+                minAbs == 0 and secOffset == 0
+            )
+            ):
```

Comment:
With if statements this big it might be better to have a separate function `date_found()` so the if statement becomes

```python
if not date_found():
```

The code gets more readable and the comment isn't really needed.

In this case you have many variables but if you make it a nested function you won't need to pass them.